src/higley_lab_to_nwb/extractors/processed_imaging_extractor.py

```diff
@@ -0,0 +1,169 @@
+
+        elif number_of_pixels == np.count_nonzero(mask):
+            self._mask = mask
+
+        else:
+            raise ValueError(
+                f"Can't reconstruct frame from pixel matrix. The total number of pixels {number_of_pixels} does "
+                f"not match the frame dimension {self._num_rows} * {self._num_columns} or the non-zero elements in "
+                f"the frame mask {np.count_nonzero(mask)}."
+            )
+
+    def _get_single_frame(self, frame_idx: int) -> np.ndarray:
+        frame = np.zeros((self._num_rows, self._num_columns))
+        i = 0
+        for r in range(self._num_rows):
+            for c in range(self._num_columns):
+                # "dff_blue" and "dff_uv" pixel matrix contains the pixel trace for only the non-zero elements of the
+                # mask (different pixel indexing)
+                if self._mask is not None and self._mask[r, c] != 0:
+                    frame[r, c] = self._pixels_matrix[frame_idx, i]
+                    i += 1
+                elif self._mask is None:
+                    frame[r, c] = self._pixels_matrix[frame_idx, i]
+                    i += 1
+        return frame
+
+    def get_video(self, start_frame=None, end_frame=None, channel: int = 0) -> np.ndarray:
+        """Get the video frames.
+
+        Parameters
+        ----------
+        start_frame: int, optional
+            Start frame index (inclusive).
+        end_frame: int, optional
+            End frame index (exclusive).
+
+        Returns
+        -------
+        video: numpy.ndarray
+            The video frames.
+        """
+
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self._num_frames
+        num_frames = end_frame - start_frame
+
+        video = np.zeros((num_frames, self._num_rows, self._num_columns))
+        i = 0
+        for r in range(self._num_rows):
+            for c in range(self._num_columns):
+                # "dff_blue" and "dff_uv" pixel matrix contains the pixel trace for only the non-zero elements of the
+                # mask (different pixel indexing)
+                if self._mask is not None and self._mask[r, c] != 0:
+                    video[:, r, c] = self._pixels_matrix[start_frame:end_frame, i]
+                    i += 1
+                elif self._mask is None:
+                    video[:, r, c] = self._pixels_matrix[start_frame:end_frame, i]
+                    i += 1
+
+        return video
+
+    def get_image_size(self) -> Tuple[int, int]:
+        return (self._num_rows, self._num_columns)
+
+    def get_num_frames(self) -> int:
+        return self._num_frames
+
+    def get_sampling_frequency(self) -> float:
+        return self._sampling_frequency
+
+    def get_num_channels(self) -> int:
+        return 1
+
+    def get_channel_names(self) -> list:
+        return self._channel_names
+
+    def get_dtype(self) -> DtypeType:
+        return self.get_frames(0).dtype
```

Comment:
I would change this to get it from `self._pixels_matrix` at init. 
e.g.
`self._dtype = self._pixels_matrix[0].dtype` 
```suggestion
        return self._dtype
```